lib/csv/table.rb

```diff
@@ -792,11 +792,48 @@ def push(*rows)
       self # for chaining
     end
 
+    # :call-seq:
+    #   table.delete(*indexes) -> deleted_values
+    #   table.delete(*headers) -> deleted_values
+    #
+    # If the access mode is <tt>:row</tt> or <tt>:col_or_row</tt>,
+    # and each argument is either an \Integer or a \Range,
+    # returns deleted rows.
+    # Otherwise, returns deleted columns data.
+    #
+    # In either case, the returned values are in the order
+    # specified by the arguments.  Arguments may be repeated.
     #
-    # Removes and returns the indicated columns or rows. In the default mixed
-    # mode indices refer to rows and everything else is assumed to be a column
-    # headers. Use by_col!() or by_row!() to force the lookup.
+    # ---
+    #
+    # Returns rows as an \Array of \CSV::Row objects:
```

Comment:
I'm a little confused here; does `delete` need the example/explanation of `by_row!` ?

Reply:
The <tt>by_row!</tt> call was not needed in the first place.  Removed.